src/model_theory/basic.lean

```diff
@@ -106,9 +126,21 @@ instance is_relational_sum [L.is_relational] [L'.is_relational] : is_relational
 instance is_algebraic_sum [L.is_algebraic] [L'.is_algebraic] : is_algebraic (L.sum L') :=
 ⟨λ n, sum.is_empty⟩
 
-instance is_relational.countable_functions [L.is_relational] :
-  nonempty (encodable (Σl, L.functions l)) :=
-⟨infer_instance⟩
+@[priority 100] instance encodable.countable [h : encodable L.symbols] :
+  L.countable :=
+⟨cardinal.encodable_iff.1 ⟨h⟩⟩
+
+@[priority 100] instance countable.countable_functions [L.countable] :
+  L.countable_functions :=
+⟨begin
+  refine lift_le_omega.1 (trans _ L.card_le_omega),
+  rw [card, symbols, mk_sum],
+  exact le_self_add,
+end⟩
+
+@[priority 100] instance encodable.countable_functions [h : encodable (Σl, L.functions l)] :
+  L.countable_functions :=
+⟨cardinal.encodable_iff.1 ⟨h⟩⟩
```

Comment:
I don't think this is a good instance. Type-class inference will basically never know `encodable (Σl, L.functions l)`, and if it does, it's easy to let it know about `L.countable_functions`.  (Feel free to keep it as lemma.)

Reply:
Alright. The case I actually want this for is `[L.is_relational]`, which does give an instance of `encodable (Σl, L.functions l)`. I can just provide that instance directly using this proof.

Reply:
Yeah, that seems fine to me.